Test target support

fredpi opened the pull request:
Closes #1 #6.

Jeehut committed:
Add support for updating dependencies of test targets

fredpi committed:
Add support for test targets with tests

fredpi committed:
Remove unnecessary groups from group 'Dependencies'

fredpi committed:
Unlink removed dependencies from frameworks build phase

Jeehut commented on Sources/AccioKit/Services/XcodeProjectIntegrationService.swift:
```diff
@@ -16,6 +16,28 @@ final class XcodeProjectIntegrationService {
+        print("Removing unnecessary groups \(groupsToRemove.compactMap { $0.name }) from group 'Dependencies' ...", level: .info)
+
+        for groupToRemove in groupsToRemove {
+            dependenciesGroup.children.removeAll { $0 == groupToRemove }
```
Here, you should load the franeworksbuildPhase of the target named like the group (if existing) and remove all frameworks within the group from the frameworks build phase.

Jeehut reviewed (changes requested):
Looks good overall, just missing the cleanup of frameworks from build phase at the commented place.

fredpi committed:
Unlink frameworks from targets that were removed

fredpi committed:
Adjust local cache logging to match shared cache logging

fredpi committed:
Delete unneeded dependencies

fredpi committed:
Only link frameworks when needed

fredpi committed:
Don't save build products into local cache if shared cache is available

fredpi committed:
Add changelog entries

fredpi committed:
Fix unlinking of removed targets not to unlink non-Accio frameworks

fredpi committed:
Remove outdated framework references; adjust logging

fredpi committed:
Use Swift 5 for package description in README

Jeehut commented on README.md:
````diff
@@ -131,7 +131,7 @@ Accio uses the [official SwiftPM manifest format](https://github.com/apple/swift
 
 ```swift
-// swift-tools-version:4.2
+// swift-tools-version:5.0
````
Could you also add this change to the `Package.swift` file created by the `init` command?

Jeehut commented on Sources/AccioKit/Services/XcodeProjectIntegrationService.swift:
```diff
@@ -16,15 +16,58 @@ final class XcodeProjectIntegrationService {
+    func unlinkAndRemoveGroupsOfUnnededTargets(keepingTargets targetsToKeep: [AppTarget]) throws {
+        guard let projectName = targetsToKeep.first?.projectName else { return }
+
+        // find groups of which the associated target is not be kept
```
"is not be kept"? Are you missing a "to" there?

Jeehut commented on Sources/AccioKit/Services/XcodeProjectIntegrationService.swift:
```diff
@@ -16,15 +16,58 @@ final class XcodeProjectIntegrationService {
+            print("Removing frameworks \(groupToRemove.children.compactMap { $0.name }) from project navigator group '\(groupName)' \(printSuffix)", level: .info)
+
+            groupToRemove.children.removeAll()
+            frameworkBuildPhase?.files.removeAll { file in groupToRemove.children.compactMap { $0.name }.contains { $0 == file.file?.name } }
```
How is `groupToRemove.children` here not empty after calling `removeAll` right one line before this?

fredpi replied:
Oh my bad 🤦‍♂️

Jeehut commented on Sources/AccioKit/Services/XcodeProjectIntegrationService.swift:
```diff
@@ -16,15 +16,58 @@ final class XcodeProjectIntegrationService {
+    }
+
+    func clearDependenciesFolder() throws {
+        let dependenciesPath = "\(workingDirectory)/\(Constants.dependenciesPath)/*"
```
Why the `/*` suffix? That's not needed afaik when running `rm -rf`.

Jeehut commented on Sources/AccioKit/Services/XcodeProjectIntegrationService.swift:
```diff
@@ -16,15 +16,58 @@ final class XcodeProjectIntegrationService {
         let dependenciesPlatformPath = "\(workingDirectory)/\(Constants.dependenciesPath)/\(platform.rawValue)"
-        let copiedFrameworkProducts: [FrameworkProduct] = try copyFrameworkProducts(frameworkProducts, to: dependenciesPlatformPath)
-        try linkFrameworks(copiedFrameworkProducts, with: appTarget, for: platform)
+        let copiedFrameworkProducts: [FrameworkProduct] = try copy(frameworkProducts: frameworkProducts, of: appTarget, to: dependenciesPlatformPath)
```
Nice catch there. 👍

Jeehut reviewed (commented):
Great work overall, please fix the commented lines and after adding the copy frameworks phase in test targets, it should be good to merge! 👍

fredpi committed:
Adjust code according to code review

fredpi committed:
Add framework copy files phase for test targets

fredpi committed:
Add Accio copy frameworks phase for test targets to fix testing

Also: Cleanup Accio run script phase when regular target gets removed from package

fredpi committed:
Fix missing use of xcodeDependenciesGroup & dependenciesPath constants.

fredpi committed:
Adjust manifest creation test to changed swift tools version

fredpi committed:
Sort dependencies group alphabetically

Jeehut commented:
Thank you for this, looks good, merging.
